packages/web/src/application/test/zoom/zoom-activity-view.test.tsx

```diff
@@ -0,0 +1,233 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { screen, waitFor } from '@testing-library/react';
+import ZoomMtgEmbedded from '@zoomus/websdk/embedded';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import application from '@application';
+
+import {
+  renderWithProviderAndCustomConsumer,
+  getElementByRegex,
+  mockDataResponse,
+  getContextInitialData,
+} from './utils';
+import { ZoomConnectionStatus } from '../../../modules/zoom/context';
```

Comment:
Shouldn't a module's context be available only to itself? Not to the entire application

Reply:
It depends.

What I see now, and haven't before, you're not using the context but rather `ZoomConnectionStatus`, so the connection status should be available in the global scope. 

Look at it this way. The application defines a global scope, in this case component `ZoomActivityView`. If the part of the global definition, in any way, is `ZoomConnectionStatus` then it should be available. If it's only a part of the implementation then it shouldn't.

In this particular case, you're testing status on an application level, not module, so it must mean that status definition is imposed by the application not the module.

Ask yourself, what is the definition and what is the implementation detail. Is status something that application expects component to respect, or the status is a means to do the expected behaviour.

Reply:
I agree for the `status`, and also for `ZoomContextType` imported in the `zoom-provider.test.tsx`, but there I import the `ZoomContext`, what about it?

Reply:
Apply the same logic. If ZoomContext is defined by the application, and it seems so, then it's not a problem it becomes a part of the global level.

Do determine if it's a part of application just consider what application expects from provider, is it context or again is a context means for something.

The real question here is, how can potentially other modules use ZoomProvider and what is envisioned use? For example:

1. One way to use it might be, ZoomProvider should eventually show enable Zoom Activity view to work. You don't test anything related to the context but the end goal.
2. The other approach is that ZoomProvider allows other modules create their own Zoom views which means they should know about the context. In this case you actually expect that context behaves in a certain way.

Since we're already half way in the 2. approach let's add context to the global scope. It makes sense that other modules might want to create their own views, this was our initial idea.